### ISSUE TITLE
Failure when publishing build scan with platforms and catalogs

### ISSUE THREAD

**Issue by ghale** [gradle/gradle#21062]
There's something going sideways when we use both platforms and catalogs together.  With the right set of dependencies, we get a failure like this when we use `--scan` to publish to scans.gradle.com or to a Gradle Enterprise instance:

```
A build scan cannot be produced as an error occurred gathering build data.
Please report this problem via https://gradle.com/help/plugin and include the following via copy/paste:

----------
Gradle version: 7.4.2
Plugin version: 3.8.1

org.gradle.api.artifacts.ResolveException: Could not resolve all dependencies for configuration ':lib:testCompileClasspath'.
        at org.gradle.api.internal.artifacts.ivyservice.ErrorHandlingConfigurationResolver.wrapException(ErrorHandlingConfigurationResolver.java:107)
        at org.gradle.api.internal.artifacts.ivyservice.ErrorHandlingConfigurationResolver$ErrorHandlingResolutionResult.getRoot(ErrorHandlingConfigurationResolver.java:214)
        at org.gradle.api.internal.artifacts.configurations.ResolveConfigurationResolutionBuildOperationResult.getRootComponent(ResolveConfigurationResolutionBuildOperationResult.java:60)
        at com.gradle.scan.plugin.internal.b.f.a.h$a.b(SourceFile:113)
        at com.gradle.scan.plugin.internal.b.f.a.h$a.a(SourceFile:66)
        at com.gradle.scan.plugin.internal.b.f.a.h.a(SourceFile:47)
        at com.gradle.scan.plugin.internal.b.f.a.g.a(SourceFile:147)
        at com.gradle.scan.plugin.internal.b.f.c.a(SourceFile:137)
        at com.gradle.scan.plugin.internal.l.a.f.a(SourceFile:12)
        at com.gradle.scan.plugin.internal.l.a$c.finished(SourceFile:154)
        at com.gradle.scan.plugin.internal.l.a.a(SourceFile:65)
        at com.gradle.scan.plugin.internal.l.l.a(SourceFile:56)
        at com.gradle.scan.plugin.internal.l.c.a(SourceFile:98)
        at com.gradle.scan.plugin.internal.l.g.a(SourceFile:51)
        at com.gradle.scan.plugin.internal.q.a$a.a(SourceFile:31)
        at com.gradle.scan.plugin.internal.q.a$a.a(SourceFile:20)
        at com.gradle.scan.plugin.internal.q.a.c(SourceFile:67)
Caused by: java.lang.RuntimeException: Problems reading data from Binary store in /Users/ghale/.gradle/.tmp/gradle1491328417675576407.bin offset 2495 exists? true
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.DefaultBinaryStore$SimpleBinaryData.read(DefaultBinaryStore.java:132)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.StreamingResolutionResultBuilder$RootFactory.lambda$create$0(StreamingResolutionResultBuilder.java:189)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.CachedStoreFactory$SimpleStore.load(CachedStoreFactory.java:101)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.StreamingResolutionResultBuilder$RootFactory.create(StreamingResolutionResultBuilder.java:187)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.StreamingResolutionResultBuilder$RootFactory.create(StreamingResolutionResultBuilder.java:159)
        at org.gradle.api.internal.artifacts.result.DefaultResolutionResult.getRoot(DefaultResolutionResult.java:50)
        at org.gradle.api.internal.artifacts.ivyservice.ErrorHandlingConfigurationResolver$ErrorHandlingResolutionResult.getRoot(ErrorHandlingConfigurationResolver.java:212)
        at org.gradle.api.internal.artifacts.configurations.ResolveConfigurationResolutionBuildOperationResult.getRootComponent(ResolveConfigurationResolutionBuildOperationResult.java:60)
        at com.gradle.scan.plugin.internal.b.f.a.h$a.b(SourceFile:113)
        at com.gradle.scan.plugin.internal.b.f.a.h$a.a(SourceFile:66)
        at com.gradle.scan.plugin.internal.b.f.a.h.a(SourceFile:47)
        at com.gradle.scan.plugin.internal.b.f.a.g.a(SourceFile:147)
        at com.gradle.scan.plugin.internal.b.f.c.a(SourceFile:137)
        at com.gradle.scan.plugin.internal.l.a.f.a(SourceFile:12)
        at com.gradle.scan.plugin.internal.l.a$c.finished(SourceFile:154)
        at com.gradle.scan.plugin.internal.l.a.a(SourceFile:65)
        at com.gradle.scan.plugin.internal.l.l.a(SourceFile:56)
        at com.gradle.scan.plugin.internal.l.c.a(SourceFile:98)
        at com.gradle.scan.plugin.internal.l.g.a(SourceFile:51)
        at com.gradle.scan.plugin.internal.q.a$a.a(SourceFile:31)
        at com.gradle.scan.plugin.internal.q.a$a.a(SourceFile:20)
        at com.gradle.scan.plugin.internal.q.a.c(SourceFile:67)
Caused by: java.lang.IllegalStateException: Corrupt serialized resolution result. Cannot find selected module (253) for constraint apiElements -> com.googlecode.libphonenumber:libphonenumber:8.12.30
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.DefaultResolutionResultBuilder.visitOutgoingEdges(DefaultResolutionResultBuilder.java:83)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.StreamingResolutionResultBuilder$RootFactory.deserialize(StreamingResolutionResultBuilder.java:239)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.DefaultBinaryStore$SimpleBinaryData.read(DefaultBinaryStore.java:130)
        ... 21 more
----------
```

I was able to boil this down to a [simple reproducible case](https://github.com/ghale/platformIssueReproducer).  What I found was that any of the following would cause the error to go away:

- Removing the version catalog and inlining the spring dependency in the platform
- Removing any one of the dependencies in the platform
- Explicitly setting the version of `mockserver-netty` in lib/build.gradle

Might be other remediating tweaks, too, that I didn't stumble on.

### Expected Behavior
Build scan should publish successfully.

### Current Behavior
An exception is thrown when publishing the build scan.

### Context
This came up in a user's build, but we were able to isolate the issue and boil it down to a simple reproducible example.

### Steps to Reproduce 
https://github.com/ghale/platformIssueReproducer

**Comment by jbartok** [comment]
Could you pls. check again with 7.5-rc-3, once it's released. It should fix it.

**Comment by ghale** [comment]
My bad for failing to update this.  We did try this and the issue is fixed with 7.5-rc-3.

**Comment by krotscheck** [comment]
@jbartok - I don't suppose this fix can be backported to the 7.4 branch?

**Comment by jbartok** [comment]
We don't usually do that, but the good news is that 7.5 should be out in a matter of days.
